lib/rspec-puppet/support.rb

```diff
@@ -15,6 +15,27 @@ def environment
       'rp_env'
     end
 
+    def stub_file_consts(example)
+      munged_facts = facts_hash(nodename(example.metadata[:type]))
+
+      if munged_facts['operatingsystem'] && munged_facts['operatingsystem'].to_s.downcase == 'windows'
+        stub_const_wrapper('PATH_SEPARATOR', ';')
+        stub_const_wrapper('ALT_SEPARATOR', "\\")
+      else
+        stub_const_wrapper('PATH_SEPARATOR', ':')
+        stub_const_wrapper('ALT_SEPARATOR', nil)
```

Comment:
Yeah, it's a nasty solution for a nasty problem (Puppet really doesn't want to do what we're forcing it to do) 😀